### PR TITLE
Bug 1311285 - Use ListClusters for updating cluster info

### PR DESCRIPTION
**Pull request by jezdez**

### CODE REVIEW

**Comment by jezdez** [comment]
@maurodoglio r?

File: atmo/clusters/jobs.py [hunk 1]
```diff
@@ -32,7 +34,58 @@ def delete_clusters():
...
+    """
+    Update the public IP address for the cluster with the given cluster ID
+    """
+    cluster = Cluster.objects.get(id=cluster_id)
+    # quick way out in case this job was called accidently
+    if cluster.master_address and not force:
+        return
+    # first get the cluster info from AWS
+    info = cluster.get_info()
+    master_address = info.get('public_dns') or ''
```
Comment (maurodoglio) [inline atmo/clusters/jobs.py]: what about using `info.get('public_dns', '')` ?

File: atmo/clusters/jobs.py [hunk 2]
```diff
@@ -32,7 +34,58 @@ def delete_clusters():
...
+
+        # don't update the state if it's equal to the already stored state
+        if info['state'] == cluster.most_recent_status:
+            continue
+
+        # run an UPDATE query for the cluster
+        cluster.most_recent_status = info['state']
+        cluster.save(update_fields=['most_recent_status'])
+
+        # in case not
```
Comment (maurodoglio) [inline atmo/clusters/jobs.py]: partial comment

File: atmo/clusters/jobs.py [hunk 3]
```diff
@@ -32,7 +34,58 @@ def delete_clusters():
...
-    for cluster in Cluster.objects.exclude(most_recent_status__in=Cluster.FINAL_STATUS_LIST):
-        cluster.update_status()
-        cluster.save()
+    """
+    Update the cluster metadata from AWS for the pending
+    clusters.
+
+    - To be used periodically.
+    - Won't update state if not needed.
+    - Will queue updateing the Cluster's public IP address if needed.
```
Comment (maurodoglio) [inline atmo/clusters/jobs.py]: s/updateing/updating

File: atmo/provisioning.py [hunk 4]
```diff
@@ -96,5 +100,21 @@ def cluster_info(jobflow_id):
     }
 
 
+def cluster_list():
+    """
+    Return a lust of cluster info with the fields
```
Comment (maurodoglio) [inline atmo/provisioning.py]: s/lust/list

File: atmo/clusters/jobs.py (hunk 1 above)
Comment (jezdez) [inline atmo/clusters/jobs.py]: That wouldn't catch an empty string of `public_dns`, which IIRC was an issue in the past

File: atmo/clusters/jobs.py [hunk 5]
```diff
@@ -32,7 +34,58 @@ def delete_clusters():
...
+    Update the public IP address for the cluster with the given cluster ID
+    """
+    cluster = Cluster.objects.get(id=cluster_id)
+    # quick way out in case this job was called accidently
+    if cluster.master_address and not force:
+        return
+    # first get the cluster info from AWS
+    info = cluster.get_info()
+    master_address = info.get('public_dns') or ''
+    # the store the public IP of the cluster if found in response
```
Comment (jezdez) [inline atmo/clusters/jobs.py]: 👍

File: atmo/clusters/jobs.py (hunk 3 above)
Comment (jezdez) [inline atmo/clusters/jobs.py]: 👍

File: atmo/clusters/jobs.py (hunk 2 above)
Comment (jezdez) [inline atmo/clusters/jobs.py]: 🖐 🤓

File: atmo/provisioning.py (hunk 4 above)
Comment (jezdez) [inline atmo/provisioning.py]: 😆

**Pull request merged**